packages/react-switch/src/components/Switch/useSwitchState.ts

```diff
@@ -46,7 +46,7 @@ export const useSwitchState = (state: SwitchState) => {
     initialState: false,
   });
   const [thumbAnimation, { setTrue: showThumbAnimation, setFalse: hideThumbAnimation }] = useBoolean(true);
-  const [renderedPosition, setRenderedPosition] = React.useState<number | undefined>(currentValue === true ? 100 : 0);
+  const [renderedPosition, setRenderedPosition] = React.useState<number | undefined>(undefined);
```

Comment:
renderedPosition does not need to start defined. If you pass in a currentValue there's a check below that looks for that before rendering the switch position.